sql/views.sql

```diff
@@ -259,5 +259,121 @@ CREATE OR REPLACE VIEW timescaledb_information.data_node AS
     LEFT OUTER JOIN LATERAL @extschema@.data_node_hypertable_info(CASE WHEN s.node_up THEN s.node_name ELSE NULL END) size ON TRUE
   GROUP BY s.node_name, s.node_up, s.owner, s.options;
 
+-- chunks metadata view, shows information about the primary dimension column
+CREATE OR REPLACE VIEW  timescaledb_information.chunks
+AS
+SELECT 
+       hypertable_schema, hypertable_name,
+       schema_name as chunk_schema , chunk_name ,
+       primary_dimension, primary_dimension_type,
+       range_start, range_end,
+       integer_range_start as range_start_integer,
+       integer_range_end as range_end_integer,
+       is_compressed,
+       chunk_table_space as chunk_tablespace,
+       node_list as data_nodes
+       from
+(
+SELECT
+    ht.schema_name as hypertable_schema,
+    ht.table_name as hypertable_name,
+    srcch.schema_name as schema_name,
+    srcch.table_name as chunk_name,
+    dim.column_name as primary_dimension,
+    dim.column_type as primary_dimension_type,
+    row_number() over(partition by chcons.chunk_id order by chcons.dimension_slice_id) as chunk_dimension_num,
```

Comment:
Would suggest to use a CTE and split it up into several queries. Will be much easier to follow and read.

Reply:
Unfortunately CTEs have performance issues. PG doesn't optimize them as expected.  That's why I rewrote the CTEs to do joins directly. (You will find the CTE version in the draft PR).

Reply:
Could you please add a comment to the code explaining that so that it is clear and nobody reverts into CTE in that case? There is already a lot of choices in the code that have no clear rationale and it is good to know if a choice is because "it seemed good at the time" or if there is a specific reason for the choice.

Reply:
added